### ISSUE TITLE
Travelling in other countrie

### ISSUE THREAD

**Issue by cvaldera** [corona-warn-app/cwa-documentation#55]
## What is missing

An explanation of what happens when someone goes to another country. Will the application interact only with other german mobile phones?
Should an international traveler install one application per country? 
## Why should it be included

Because there are people who travel a lot in different countries for personal / professional reasons

**Comment by aeisenbach** [comment]
This "roaming scenario" is being considered, but will not be part of the first release of the application.

**Comment by egandro** [comment]
> This "roaming scenario" is being considered, but will not be part of the first release of the application.

How do you make sure, that we don't violate the law or foreign countries?

The tracking App might be illegal e.g. in Russia or Arabian countries.

**Comment by IXArcanumXI** [comment]
You have to make sure you don't break the laws in foreign countries @egandro . If Russia bans the app, it is your responsibility to deinstall it prior entering the country. Same story as "Blitzer-Apps" - might be legal if you use it in a foreign country - if you use it in Germany, it's your fault, not the developers fault.

**Comment by Leseratte10** [comment]
@egandro That is not the task for the developer. The developer is offering this software in Germany. If you use it and travel to another country, YOU are responsible for making sure that any app you use is legal in that country.

**Comment by Mikeichhalt** [comment]
Isn't every app that uses the Google/Apple Api compatible to each other by default?

**Comment by MikeJayDee** [comment]
> Isn't every app that uses the Google/Apple Api compatible to each other by default?

Yes in one way: you can install several apps that can access the API. So if I travel from Germany to Denmark, I install the Danish app before I arrive in Denmark, and leave it installed for two weeks after my return to Germany. I will then get warned by the Danish about any contacts in Denmark that later had a positive test.

But also no in a sense: if I get infected and am infectious already in Denmark but only take a test in Germany, there will not be a simple way for the test result to be pushed to the Danish server.

**Comment by keugens** [comment]
> ... you can install several apps ...

Is there a good reason for that? Not that I know. Therefore, there is probably the effect that many potential virus spreaders will not participate.

**Comment by treysis** [comment]
> This "roaming scenario" is being considered, but will not be part of the first release of the application.

If you don't implement the roaming scenario from the get-go - i.e. to have interoperable specifications - chances are it will only get more difficult downstream.

**Comment by LukasMasuch** [comment]
Cross-country interoperability is already worked on in collaboration with Apple/Google and many other countries. All apps that use the Exposure Notification API are already compatible at the Bluetooth level. To enable full interoperability, the diagnostic keys must be exchanged between national backends. Once a common API specification for key exchange is implemented, users won't need to install additional national apps when traveling in interoperable countries. This is currently work in progress, and additional information will be shared soon. In the meantime, [DP3T's interoperability paper](https://drive.google.com/file/d/1mGfE7rMKNmc51TG4ceE9PHEggN8rHOXk/edit) provides a good overview of this topic.

**Comment by keugens** [comment]
@LukasMasuch @SebastianWolf-SAP
It seems the paper you mentioned is protected against copy and print, which probably does not comply to the stated Apache Licence. Could you please clarify?

**Comment by raethlein** [comment]
@keugens Since it is an externally-linked document, I guess it does not fall under the Apache license of this repository.

**Comment by keugens** [comment]
**Decentralized Proximity Tracing, Interoperability Specification; 18 May 2020; by DP3T**
[https://github.com/DP-3T/documents/blob/master/DP3T%20-%20Interoperability%20Decentralized%20Proximity%20Tracing%20Specification%20(Preview).pdf](url)

**Interoperability of decentralized proximity tracing systems across regions; 15 May 2020; by Accenture/Austria and others**
[https://drive.google.com/file/d/1mGfE7rMKNmc51TG4ceE9PHEggN8rHOXk/edit](url)
Note: This document is protected against copy and print and probably cited wrongly as _DP3T's interoperability paper_ in this thread.

**Comment by LukasMasuch** [comment]
@keugens the DP3T team worked on two papers about interoperability: 1) A [conceptual white paper](https://drive.google.com/file/d/1mGfE7rMKNmc51TG4ceE9PHEggN8rHOXk/edit) written as a joint effort between DP3T and many other companies. You can find proof of DP3T involvement [here](https://twitter.com/carmelatroncoso/status/1257576532308037632) or [here](https://twitter.com/mikarv/status/1257714842892369920). I don't know why it is currently protected against copy and print, but it is not under Apache 2.0 license since it is not an official document from the CWA project. 2) A [technical specification](https://github.com/DP-3T/documents/blob/master/DP3T%20-%20Interoperability%20Decentralized%20Proximity%20Tracing%20Specification%20(Preview).pdf) that contains specific proposals for API and system design.

**Comment by amandadebler** [comment]
Will the app be available in non-German app stores? I'm a resident of Germany, but my phone is still connected to the US app store, because I need access to the apps for some US banks.

**Comment by kolAflash** [comment]
@LukasMasuch 

> > Isn't every app that uses the Google/Apple Api compatible to each other by default?
> 
> Yes in one way: you can install several apps that can access the API. So if I travel from Germany to Denmark, I install the Danish app before I arrive in Denmark, and leave it installed for two weeks after my return to Germany. I will then get warned by the Danish about any contacts in Denmark that later had a positive test.

Are you sure that will work?
This issue looks differently: #329 _Incompatibility with the App Immuni_
Quote: _only 1 app can connect to their Exposure Notifications API on each device._
&nbsp;

> Cross-country interoperability is already worked on [...]

Can you tell what's the rough status of this?
Which countries have apps with possibly compatible architectures and which other app projects are you talking too?

Currently I only know two other apps which might be compatible:
- Italy (open source): https://github.com/immuni-app
- Denmark (closed source): https://smittestop.dk

As far as I heard the French app stores data centralizes and so it's incompatible (technically and regarding data protection policy).

**Comment by akuckartz** [comment]
This issue was raised 15 May 2020 - more than a month ago - and is still not resolved. Why not?

**Comment by cfritzsche** [comment]
I assume the reason is that european alignment is not quick to achieve. The european commission has just released its guidelines for interoperability a week ago: https://ec.europa.eu/health/sites/health/files/ehealth/docs/mobileapps_interoperabilityspecs_en.pdf

**Comment by MikeJayDee** [comment]
> @LukasMasuch
> 
> > > Isn't every app that uses the Google/Apple Api compatible to each other by default?
> > 
> > 
> > Yes in one way: you can install several apps that can access the API. So if I travel from Germany to Denmark, I install the Danish app before I arrive in Denmark, and leave it installed for two weeks after my return to Germany. I will then get warned by the Danish about any contacts in Denmark that later had a positive test.
> 
> Are you sure that will work?
> This issue looks different: #329 _Incompatibility with the App Immuni_
> Quote: _only 1 app can connect to their Exposure Notifications API on each device._

Yes and no. It turns out that only one app can be active at any point in time. If you have the German app active on one day, the Italian on the next and so on, you would still be warned from both (with an additional delay). When I wrote my original comment, the API was not yet published yet and there was no indication that only one app can be active at any point in time.

> Which countries have apps with possibly compatible architectures and which other app projects are you talking too?
> 
> Currently I only know two other apps which might be compatible:
> 
> * Italy (open source): https://github.com/immuni-app
> * Denmark (closed source): https://smittestop.dk

Out of my head, the following countries use the Exposure Notification API (probably not complete):

**Active:**
Latvia - Apturi Covid
Italy - Immuni
Switzerland - SwissCovid (Beta)
Denmark - Smittestop
Poland - ProteGO
Uruguay - Coronavirus-UY
Japan - COCOA

**Announced** (in no particular order, definitely not complete):
Ireland (possibly next week already)
Estonia
Austria
Spain
Portugal
Belgium
Netherlands
Canada (in July)
UK (only in late 2020)



> 
> As far as I heard the French app stores data centralizes and so it's incompatible (technically and regarding data protection policy).
